overlay_new.py

```diff
@@ -0,0 +1,197 @@
+from abc import ABC, abstractmethod
+from time import time
+
+from canvas import Canvas, Colour
+from data import Data
+from overlay import Overlay
+from textwrap import wrap
+from typing import Callable, Tuple
+
+class Drawable(ABC):
+    @abstractmethod
+    def draw_base(self, canvas: Canvas):
+        pass
+    @abstractmethod
+    def draw_data(self, canvas: Canvas, data: Data):
+        pass
```

Comment:
I am still getting used to abstract base classes. Why is this important define this class as an ABC? Is it just because the methods are used when inherited from multiple other classes?

Reply:
ABC is used for the parent class which prevents child classes from being used until all abstract methods are overridden. It can be important to define this class as an ABC as it ensures a script using this parent class in the future will run properly and output as intended. 

As far as I know, it doesn't run until ALL abstract methods have been overridden, even if you aren't using them. 

More info here: https://docs.python.org/3/library/abc.html#abc.abstractmethod
Hope this helps!

Reply:
Take a look at the `drawables` list in the overlay itself. If we know that the `drawables` list contains only objects which implement `Drawable`, then we know that it is always safe to call `draw_base` and `draw_data` on any and every object inside `drawables`. This wouldn't be the case if `Drawable` subclasses weren't forced to implement these functions!

Of course, there's nothing stopping a person from putting something that doesn't implement `Drawable` into the list, but if we're able to assume that no one's done something silly, it makes working with the `drawables` list much neater. In strongly typed languages, lists usually only can contain objects of a single type, meaning that we could enforce that everything in `drawables` is a `Drawable`!

Reply:
Good question @Blake-Haydon and answers @nix-chu @hallgchris. Pretty much said everything that I wanted to. 

Let us know if you're still confused!